pkg/server/bundle/datastore/wrapper_test.go

```diff
@@ -0,0 +1,31 @@
+package datastore
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/spiffe/spire/test/fakes/fakedatastore"
+	"github.com/spiffe/spire/test/fakes/fakepubmanager"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWithBundlePublisher(t *testing.T) {
+	pubManager := fakepubmanager.New()
+
+	ds := WithBundleUpdateCallback(fakedatastore.New(t), pubManager.BundleUpdated)
```

Comment:
Now that this uses a callback, i don't think you need the fakepubmanager anymore. You can just assert that the callback was invoked.